File: ios/closer/EncryptedDownloaderBridge.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(EncryptedDownloaderBridge, NSObject)

RCT_EXTERN_METHOD(downloadAndDecrypt:(NSDictionary *)options
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/closer/EncryptedDownloaderBridge.swift
import Foundation
import React

@objc(EncryptedDownloaderBridge)
class EncryptedDownloaderBridge: NSObject {
    
    @objc(downloadAndDecrypt:withResolver:withRejecter:)
    func downloadAndDecrypt(_ options: NSDictionary, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        EncryptedDownloader.downloadAndDecrypt(options as! [String: Any], resolver: resolve, rejecter: reject)
    }
    
    @objc static func requiresMainQueueSetup() -> Bool {
        return false
    }
}